src/wallet/components/WalletDropdownFundLink.tsx

```diff
@@ -14,69 +10,39 @@ export function WalletDropdownFundLink({
   rel,
   target,
   text = 'Fund wallet',
+  fundingUrl,
 }: WalletDropdownFundLinkReact) {
-  const [fundingUrl, setFundingUrl] = useState('');
-
-  const iconSvg = useIcon({ icon });
-
-  useEffect(() => {
-    const currentURL = window.location.href;
-    const tabName = document.title;
-    const url = `http://keys.coinbase.com/fund?dappName=${encodeURIComponent(
-      tabName,
-    )}&dappUrl=${encodeURIComponent(currentURL)}&version=${encodeURIComponent(
-      version,
-    )}&source=onchainkit`;
-    setFundingUrl(url);
-  }, []);
-
-  const handleClick = useCallback(
-    (e: React.MouseEvent) => {
-      e.preventDefault();
-      const { width, height } = getWindowDimensions(popupSize);
-
-      const left = Math.round((window.screen.width - width) / 2);
-      const top = Math.round((window.screen.height - height) / 2);
-
-      const windowFeatures = `width=${width},height=${height},resizable,scrollbars=yes,status=1,left=${left},top=${top}`;
-      window.open(fundingUrl, target, windowFeatures);
-    },
-    [fundingUrl, popupSize, target],
-  );
-
-  const overrideClassName = cn(
-    pressable.default,
-    'relative flex items-center px-4 py-3 w-full',
-    className,
-  );
-
-  const linkContent = useMemo(
-    () => (
-      <>
-        <div className="-translate-y-1/2 absolute top-1/2 left-4 flex h-[1.125rem] w-[1.125rem] items-center justify-center">
-          {iconSvg}
-        </div>
-        <span className={cn(themeText.body, 'pl-6')}>{text}</span>
-      </>
-    ),
-    [iconSvg, text],
-  );
-
-  if (openIn === 'tab') {
+  const defaultFundingUrl = useGetFundingUrl();
+
+  // If the fundingUrl prop is undefined, fallback to the default funding URL
+  const fundingUrlToRender = fundingUrl ?? defaultFundingUrl?.url;
+  const popupHeightOverride = fundingUrl
+    ? undefined
+    : defaultFundingUrl?.popupHeight;
+  const popupWidthOverride = fundingUrl
+    ? undefined
+    : defaultFundingUrl?.popupWidth;
```

Comment:
const popupHeightOverride = fundingUrl?.popupHeight would end up the same right?

Reply:
@alessey Na, even if they provide a fundingUrl prop we still fetch the defaultFunding URL (can't conditionally call a hook) so we'd risk overriding the popup size when the use provides a fundingUrl prop.

Reply:
true, you could pass in fundingUrl to your useFundingUrl to short circuit the whole thing, which would reduce this component to a single line hook call and may remove the need for it.

Reply:
Yeah but that would be a really weird interface i.e. `useFundingUrl(fundingUrl: string): string`. Why would you pass a fundingUrl to a hook to get a fundingUrl? This might not be super pretty, but from an interface perspective I think it makes the most sense.

Reply:
true, with those names it would be odd, the purpose would be ensuring fundingUrl is a consistent data type.

Reply:
@alessey can you think of better names? Even with other names, the point of the param is to basically tell the hook to do nothing because we can't conditionally call the hook. Do you think this is a blocker for the PR?

Reply:
useFundingUrlConfig could work.  I do like the idea of being able to ditch this wrapper component and simplify the override logic, but no, I do not think this is a blocker.